### PR TITLE
feat(temple): implement tier upgrade calculator

### PR DESCRIPTION
**Pull request by polaz**
## Summary
- Implement TierCalculator engine for calculating effective room tiers based on adjacency conditions, generator power coverage, and special room mechanics
- Add power coverage calculation using Manhattan distance from generators (T1=3, T2=4, T3=5 cells)
- Integrate with AdjacencyAnalyzer for upgrade requirement evaluation
- Support medallion upgrades with additive tier bonuses (capped at T3)
- Handle special rooms: Sacrificial Chamber (sacrifice count) and Spymaster (assassinate count)
- Provide simulation capabilities for placement and upgrade what-if analysis

## Test plan
- [x] 45 unit tests covering all functionality
- [x] Power coverage calculation tests (all tiers, overlapping, boundaries)
- [x] Tier calculation tests (achievable tiers, adjacency, power requirements)
- [x] Special room tests (Sacrificial Chamber, Spymaster)
- [x] Medallion upgrade tests (bonus application, T3 cap, stacking)
- [x] Simulation tests (placement, upgrades, affected rooms detection)
- [x] Edge cases (T1-only rooms, reason formatting)
- [x] Build passes
- [x] All 175 temple-related tests pass

Closes https://github.com/polaz/pob2-web/issues/164

### CODE REVIEW

File: src/engine/temple/TierCalculator.ts [hunk 1]
```diff
@@ -0,0 +1,724 @@
...
+
+    // Apply medallion upgrades if present
+    const medallionKey = `${x},${y}`;
+    const medallionBonus = context.medallionUpgrades?.get(medallionKey) ?? 0;
+
+    // Effective tier is current stored tier + medallion bonus, capped at achievable or T3
+    const baseTier = room.tier;
+    const effectiveTier = Math.min(
+      baseTier + medallionBonus,
+      3 // Max tier is always T3, even with medallions
```
[the remark this answers is not in the document]
Comment (polaz) [reply src/engine/temple/TierCalculator.ts]: Added `MAX_TEMPLE_TIER = 3` constant and updated all tier cap comparisons to use it.
